File: stacks_and_queues/daily_temperatures.js
/**
 * @param {number[]} temperatures
 * @returns {number[]}
 */
var dailyTemperatures = function(temperatures)
{
    let stack = [];
    let answer = new Array(temperatures.length).fill(0);

    for (let i = 0; i < temperatures.length; i++)
    {
        while (stack.length
               && temperatures[stack[stack.length - 1]] < temperatures[i])
        {
            const j = stack.pop();
            answer[j] = i - j;
        }

        stack.push(i);
    }

    return answer;
};

File: stacks_and_queues/daily_temperatures.py
def dailyTemperatures(temperatures: List[int]) -> List[int]:
    stack = []
    answer = [0] * len(temperatures)

    for i in range(len(temperatures)):
        while stack and temperatures[stack[-1]] < temperatures[i]:
            j = stack.pop()
            answer[j] = i - j
        
        stack.append(i)
    
    return answer